### PR TITLE
finalupdate_HW18_SolarPanelShopProject

### PR DESCRIPTION
**Pull request by Svitlytsya**

### CODE REVIEW

File: AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs [hunk 1]
```diff
@@ -88,19 +94,57 @@ public void AddAndRemoveJinkoSolarProduct545FromBusket()
                  JinkoSolarCheckbox.Should.BeChecked().
                  JinkoSolarProducts.Count.WaitTo.BeLess(countBefore);
```
[the remark this answers is not in the document]
Comment (vldemyan) [reply AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs:95]: remove duplicate [Test, Description("Verify that product can be added and removed from the Shopping Cart")]
There is no need to have 2 complete copy/paste just with [0] / [1] difference. If you want to check different products with the same test scenario, it should be test with parameters which is executed for index = 0 and index = 1, but not to copy/paste entire tests just to use another index

File: AtataUI_SolarPanelShop/PageObjects/SolarPanelPage.cs [hunk 2]
```diff
@@ -34,6 +43,15 @@ public class ProductItem : Control<_>
 
         }
 
-    }
+        [FindByXPath("//h4[contains(@class,'block-title')]")]
+        public Text<_> ProductTitle { get; private set; }
 
+        [FindByXPath("//td[contains(@class, 'col s12 m7 l6 xl4') and contains(@class, 'spec-title')]")]
```
[the remark this answers is not in the document]
Comment (Svitlytsya) [reply AtataUI_SolarPanelShop/PageObjects/SolarPanelPage.cs]: Thanks. Done.

File: AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs [hunk 3]
```diff
@@ -88,19 +94,57 @@ public void AddAndRemoveJinkoSolarProduct545FromBusket()
...
+        [Test]
+        public void VerifyPowerInProductDetailsJinkoSolar455Product()
+        {
+            var page = Go.To<SolarPanelPage>().
+                FilterButton.Click().
+                JinkoSolarCheckbox.ScrollTo();
+            var countBefore = page.JinkoSolarProducts.Count;
+            page.JinkoSolarCheckbox.Script.Click().
+                JinkoSolarCheckbox.Should.BeChecked().
+                JinkoSolarProducts.Count.WaitTo.BeLess(countBefore);
```
[the remark this answers is not in the document]
Comment (Svitlytsya) [reply AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs]: Thanks. Done.

File: AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs [hunk 4]
```diff
@@ -10,16 +10,16 @@
 namespace AtataUI_SolarPanelShop.Tests
 {
     [Description(" Verify that catalog filter works correctly")]
```
[the remark this answers is not in the document]
Comment (Svitlytsya) [reply AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs]: Thanks. Done.

File: AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs [hunk 5]
```diff
@@ -10,16 +10,16 @@
...
-    public sealed class CatalogFilterTests: UITestFixture
+    public sealed class CatalogFilterTests : UITestFixture
     {
         [Test, Description("Verify FilterButton is enabled")]
-         public void VerifyFilterButtonIsEnabled() 
-         {
+        public void VerifyFilterButtonIsEnabled()
```
[the remark this answers is not in the document]
Comment (Svitlytsya) [reply AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs:16]: Ok, thanks.

File: AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs [hunk 6]
```diff
@@ -48,37 +48,43 @@ public void VerifyFilterByJinkoSolarBrandFiltered()
...
                         JinkoSolarCheckbox.Script.Click().
                                   JinkoSolarCheckbox.Should.BeChecked();
-                                  foreach (var product in page.JinkoSolarProducts)
-                                  {
-                                        product.Title.Should.BePresent();
-                                        product.Title.Should.Contain("Jinko Solar");
-                                  }
+            foreach (var product in page.JinkoSolarProducts)
+            {
+                product.Title.Should.BePresent();
```
[the remark this answers is not in the document]
Comment (Svitlytsya) [reply AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs]: Thanks. Done.

File: AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Svitlytsya) [reply AtataUI_SolarPanelShop/Tests/SolarPanelTests.cs]: I understood..;)) i jus trained a little...than you very much..!)

**Review by vldemyan — APPROVED** [review]

**Pull request merged**